core/src/main/scala/com/yahoo/maha/core/registry/Registry.scala

```diff
@@ -21,23 +21,23 @@ import scala.collection.{SortedSet, mutable}
 class RegistryBuilder{
 
   private var publicDimMap: Map[(String, Int), PublicDimension] = Map()
-  private var publicFactMap: Map[(String, Int), PublicFact] = Map()
+  private var publicFactMap: Map[(String, Int), PublicFactTable] = Map()
   private var dimColToKeySetMap: Map[String, Set[String]] = Map()
   private var dimColToDimensionSetMap: Map[String, Set[(String, Int)]] = Map()
 
-  def register(fact: PublicFact): RegistryBuilder = {
```

Comment:
we should not be replacing Trait PublicFact With the Derived class PublicFactTable,


https://github.com/yahoo/maha/blob/c81d62dbdf1fa11ac355fc096bcacbaf24e16360/core/src/main/scala/com/yahoo/maha/core/fact/Fact.scala#L1660